### PR TITLE
Adds mouse handlers to x and y axis

### PR DESCRIPTION
**Pull request by krissalvador27**

### CODE REVIEW

File: src/XYPlot.js [hunk 1]
```diff
@@ -55,17 +39,14 @@ function getMouseOptions(
   const xValue = !_.inRange(
     innerX,
     0,
-    chartSize.width /* + padding.left + padding.right */
+    chartSize.width + marginLeft + marginRight
```
Comment (joshbDev) [inline src/XYPlot.js]: why was this commented out? does this work as expected?

File: src/XAxis.js [hunk 2]
```diff
@@ -3,13 +3,34 @@ import PropTypes from "prop-types";
...
 import { sumMargins } from "./utils/Margin";
-import { getTickDomain } from "./utils/Scale";
+import { getTickDomain, inferScaleType, invertPointScale } from "./utils/Scale";
 import xyPropsEqual from "./utils/xyPropsEqual";
 import XAxisLabels from "./XAxisLabels";
 import XAxisTitle from "./XAxisTitle";
 import XGrid from "./XGrid";
 import XTicks from "./XTicks";
 
+function getMouseOptions(event, xScale) {
```
Comment (joshbDev) [inline src/XAxis.js]: you're using this twice in two places, go ahead and add this to a new `utils` file

File: src/XYPlot.js (hunk 1 above)
Comment (krissalvador27) [reply src/XYPlot.js]: Investigating it more to make sure I don't break anything, but basically the margins are added to the chart and should be included as the overall width. Looking into yValue and if marginTop and marginBottom should also be included when thinking of the height.

Comment (krissalvador27) [reply src/XYPlot.js]: Will remove adding of margins 👍 Tested it out on XYPlot and I think it's fine without adding this to the "range" that a mouse position should be in

**Pull request merged**